components/Elevation/tests/unit/MDCMaterialElevationRespondingTests.m

```diff
@@ -0,0 +1,387 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialElevation.h"
+
+#import <XCTest/XCTest.h>
+
+/**
+ Used for testing @c UIViews that conform to @c MDCElevation and do not respond to @c
+ mdc_overrideBaseElevation.
+ */
+@interface MDCConformingMDCElevationView : UIView <MDCElevation>
+@property(nonatomic, assign, readonly) CGFloat mdc_currentElevation;
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)(CGFloat elevation);
+@property(nonatomic, assign) CGFloat elevation;
+@end
+
+@implementation MDCConformingMDCElevationView
+
+- (CGFloat)mdc_currentElevation {
+  return self.elevation;
+}
+
+@end
+
+/**
+ Used for testing @c UIViewControllers that conform to @c MDCElevation and do not respond to @c
+ mdc_overrideBaseElevation.
+ */
+@interface MDCConformingMDCElevationViewController : UIViewController <MDCElevation>
+@property(nonatomic, assign, readonly) CGFloat mdc_currentElevation;
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)(CGFloat elevation);
+@property(nonatomic, assign) CGFloat elevation;
+@end
+
+@implementation MDCConformingMDCElevationViewController
+
+- (CGFloat)mdc_currentElevation {
+  return self.elevation;
+}
+
+@end
+
+/**
+ Used for testing @c UIViews that conform to @c MDCElevation and do respond to @c
+ mdc_overrideBaseElevation.
+ */
+@interface MDCConformingMDCElevationOverrideView : UIView <MDCElevation>
+@property(nonatomic, assign, readwrite) CGFloat mdc_overrideBaseElevation;
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)(CGFloat elevation);
+@property(nonatomic, assign, readonly) CGFloat mdc_currentElevation;
+@property(nonatomic, assign) CGFloat elevation;
+@end
+
+@implementation MDCConformingMDCElevationOverrideView
+
+- (CGFloat)mdc_currentElevation {
+  return self.elevation;
+}
+
+@end
+
+@interface MDCConformingMDCElevationOverrideViewSubclass : MDCConformingMDCElevationOverrideView
+
+@end
+
+@implementation MDCConformingMDCElevationOverrideViewSubclass
+
+- (CGFloat)mdc_overrideBaseElevation {
+  if (self.superview) {
+    if ([self.superview conformsToProtocol:@protocol(MDCElevation)]) {
+      UIView<MDCElevation> *superview = (UIView<MDCElevation> *)self.superview;
+      return superview.mdc_currentElevation + superview.mdc_baseElevation +
+             self.mdc_currentElevation;
+    }
+    return self.superview.mdc_baseElevation + self.mdc_currentElevation;
```

Comment:
self.superview.mdc_baseElevation + self.superview.mdc_currentElevation ?

Reply:
Addressed in other comment.